Core/Inc/utils.h

```diff
@@ -27,14 +45,45 @@
 */
 #define fequals(a, b) (fabs(a - b) < EPSILON) ? (true) : (false)
 
-// Macro that retries a SPI TX/RX if it fails
-#define SPIRTRY(fn, hspi, ...) \
-	for (int i = 0; i < 2; i++) \
+/*!
+  @brief    Carries out a HAL SPI command and blocks the executing task until SPI complete
+  @param    fn          HAL SPI function to execute
+  @param    hspi        SPI bus handle for transaction
+  @param    timeout     Max duration of task blocking state
+  @param    status      U32 status variable that will be set according to the SPI transaction state on completion
+  @param    callingFunc Name of the function that called SPI_TRANSMIT for debugging purposes
+*/
+#define SPI_TRANSMIT(fn, hspi, timeout, status, callingFunc, ...) \
```

Comment:
Also would it be a bit cleaner to have the following format? 
`SPI_TRANSMIT(hspi, timeout, status, callingFunc, fn, ...)`

Reply:
Hmmm also I'm wondering if having the `callingFunc` as an argument might cause problems or be difficult to read. You do a good job documenting it but it might get confusing for people to see "writeRegister" in a SPI call

Reply:
Actually nvm I think I like it